tests/test_intervaldata.py

```diff
@@ -0,0 +1,196 @@
+        default_partial_inclusion=False,
+        values=[random.random() for _ in range(n)],
+    )
+
+
+@pytest.fixture()
+def intervaldata4():
+    """Creates an IntervalData object with n sequential intervals lasting 1 second each, with values."""
+    n = random.randint(50, 150)
+    return IntervalData(
+        label="intervaldata1",
+        name="IntervalData 2",
+        unit=None,
+        category="dummy",
+        intervals=list(zip(range(n), range(1, n + 1), strict=True)),
+        default_partial_inclusion=False,
+        values=np.array([random.random() for _ in range(n)]),
+    )
+
+
+def test_post_init(intervaldata1: IntervalData) -> None:
+    assert isinstance(intervaldata1.intervals, list)
+    assert all(isinstance(interval, Interval) for interval in intervaldata1.intervals)
+
+
+def test_len(intervaldata1: IntervalData) -> None:
+    assert len(intervaldata1) == len(intervaldata1.intervals)
+
+
+def test_has_values(
+    intervaldata1: IntervalData,
+    intervaldata2: IntervalData,
+    intervaldata3: IntervalData,
+    intervaldata4: IntervalData,
+) -> None:
+    assert not intervaldata1.has_values
+    intervaldata1.values = []
+    assert intervaldata1.has_values
+    intervaldata1.values = None
+
+    assert not intervaldata2.has_values
+    assert intervaldata3.has_values
+    assert intervaldata4.has_values
+
+
+def test_index_slicing(intervaldata1: IntervalData) -> None:
+    _sliced_copy = intervaldata1._sliced_copy(0, 10, newlabel="sliced_copy")  # noqa: SLF001
+    assert len(_sliced_copy) == 10
+    sliced_copy = intervaldata1[:10]
+    assert _sliced_copy == sliced_copy
+
+    assert len(intervaldata1[0]) == 1
+
+
+def test_select_by_time(intervaldata1: IntervalData, intervaldata2: IntervalData) -> None:
+    assert intervaldata1.t[:1] == intervaldata1[:1]
+
+    assert len(intervaldata1.t[:1]) == 1
+    assert len(intervaldata1.t[:0.5]) == 1
+
+    assert len(intervaldata2.t[:1]) == 1
+    assert len(intervaldata2.t[:0.5]) == 0
+
+    assert len(intervaldata1.t[1:2]) == 1
+    assert len(intervaldata2.t[1:2]) == 1
+
+    assert len(intervaldata1.t[1.5:2.5]) == 2
+    assert len(intervaldata2.t[1.5:2.5]) == 0
+
+    assert len(intervaldata1.t[1.5:3]) == 2
+    assert len(intervaldata2.t[1.5:3]) == 1
+
+    assert len(intervaldata1.t[1.5:3.5]) == 3
+    assert len(intervaldata2.t[1.5:3.5]) == 1
+
+    assert len(intervaldata1.t[2:3.5]) == 2
+    assert len(intervaldata2.t[2:3.5]) == 1
+
+    sliced_copy = intervaldata1.t[2.5:3.5]
+    assert sliced_copy.intervals[0].start_time == 2.5
+    assert sliced_copy.intervals[0].end_time == 3
+    assert sliced_copy.intervals[1].start_time == 3
+    assert sliced_copy.intervals[1].end_time == 3.5
+
+    sliced_copy = intervaldata1.t[-10:-1]
+    assert len(sliced_copy) == 0
+
+    sliced_copy = intervaldata1.t[-10:3]
+    assert len(sliced_copy) == 3
+
+    sliced_copy = intervaldata1.t[:]
+    assert len(sliced_copy) == len(intervaldata1)
```

Comment:
they should be considered equal, not just in length

```suggestion
    assertsliced_copy == intervaldata1
```